lib/puppet/file_serving/metadata.rb

```diff
@@ -118,6 +118,10 @@ def collect(source_permissions = nil)
     when "link"
       @destination = Puppet::FileSystem.readlink(real_path)
       @checksum = ("{#{@checksum_type}}") + send("#{@checksum_type}_file", real_path).to_s rescue nil
+    when "fifo"
+      @checksum = ("{#{@checksum_type}}") + send("#{@checksum_type}_file", real_path).to_s
+    when "socket"
+      @checksum = ("{#{@checksum_type}}") + send("#{@checksum_type}_file", real_path).to_s
```

Comment:
It looks like the logic around these cases is the same, so maybe just updating the `when "file"` file would work? Something like `when "file", "fifo", "socket"` ...